### ISSUE TITLE
"stateless" should be a configuration option

### ISSUE THREAD

**Issue by Bilge** [knpuniversity/oauth2-client-bundle#271]
Currently the only way to set a client as stateless is to call `OAuth2ClientInterface::setAsStateless()`. However, it should seldom be the case that this setting should change at runtime, and in fact will usually be constant from client creation time. Ergo this should be set at configuration time rather than runtime, however no such facility to do so currently exists.

Proposed configuration structure:

```yaml
knpu_oauth2_client:
    clients:
        foo:
            type: generic
            stateless: true
```

**Comment by weaverryan** [comment]
Hi @Bilge!

There is a `use_state` option that you can put under each client, which, if set to `false` should trigger that same setter to be called - https://github.com/knpuniversity/oauth2-client-bundle/blob/8ede9725f8b0d9b6ef5c93f991da8a205d911d5f/src/DependencyInjection/KnpUOAuth2ClientExtension.php#L289-L291

You can see this in the docs: https://github.com/knpuniversity/oauth2-client-bundle#configuration

Is this not working for you, or is it not working for the `generic` type?

Cheers!

**Comment by Bilge** [comment]
Interestingly it is not mentioned in [the part of the docs I was looking at](https://github.com/knpuniversity/oauth2-client-bundle#2-configuration).

**Comment by weaverryan** [comment]
You're right! Fixed in #272
